CONTRIBUTING.md

```diff
@@ -325,6 +326,84 @@ if you change the history (rebase, squash, amend), and use git push --force to
+
+**Routes** in Sinatra are kind of like a combination of routes and controllers in Rails.
+In a route, you can specify endpoints (like `exercism.io/login`) and define how the app should respond (i.e. accept some credentials and log the user in or handle errors accordingly).
+Routes are located in `app/routes.rb`. If you add a route, update the list of existing routes in this file (the convention is `ClassName: 'filename'`).
+Routes are unique in that you'll also have to update the main `app.rb` with the new route info (the convention is `Routes::ClassName`).
+
+**Views** are a collection of templates that create html.
+Templates are an awesome way to have embedded Ruby in your views so that you can serve up information,
+iterate over items without repeating yourself (like creating tables of data), etc.
+The templating engine that exercism uses is [ERB](https://en.wikipedia.org/wiki/ERuby).
+
+#### Client and Frontend (JavaScript, Custom Directives, Bootstrap, etc)
+The client side JS is a bit... spread out. There are two different JS related directories: `client/` and `frontend/`.
+Both have angular elements but `client/` is in JS and `frontend/` is mostly CoffeeScript.
+
+**Frontend** is where Exercism stores all of it's production client side code.
+Exercism doesn't have much JS (outside of bootstrap) but `frontend/` does handle comment threads on submissions, markdown preview on comments, submission code (expanding and contracting), etc. If you suspect that there is frontend wizardry happening and you haven't seen it in bootstrap, most likely it's in here.
+
+**Client** contains the code for the `comment-box` angular directive (custom html tag). Exercism doesn't use `comment-box` so this section might go away...
+
+#### DB (Migrations)
+To create a migration you can use `rake db:generate:migration name=the_name_of_your_migration`.
+That will create a new migration in `db/migrate`.
+For help, look at the other migrations or at docs for ActiveRecord migrations.
+If you add, remove or rename database columns, you will most likely have to update the seeds as well.
+Exercism pulls seed data from another repo found [here](https://github.com/exercism/seeds).
+The seed data provides placeholder information (like users, exercise submissions, comments, etc) to help with development.
+
+#### Lib (DB Configurations, Application Logic, Models, Rake Tasks)
+**Application Logic** can be anything that is used for the backend application (and, possibly, shared with the frontend as well) can live here.
+Maybe a library for GitHub OAuth, configurations for a Markdown parser and a syntax highlighter, etc.
+
+**Models** are scaffolds for how data is stored and what it can do.
+
+#### Test
+Tests follow the organization of the app. For instance, if you're writing a new route, it would go in `test/app`. You could write a test post or get and assert that you get back data that you expect. Read more about tests [here](#test-order).
+
+#### X
```

Comment:
Do you mean like pulling the markdown parser out into `exercism/` and so that it wouldn't be in `app/`?